app/models/piece.rb

```diff
@@ -5,7 +5,52 @@ class Piece < ActiveRecord::Base
   validates :color, inclusion: { in: %w(black white) }
   validates :type, inclusion: { in: %w(Pawn Rook Bishop Knight King Queen) }
 
-end
+  private
+
+  #ALL MOVE CHECKS ASSUME WHITE PLAYER IS AT BOTTOM OF 2-D ARRAY/BOARD, AND BLACK PLAYER IS ON TOP.  NO IMPACT ON VIEW. -Chad
+
+  def x_distance(x) #calculates how many tiles piece has moved on x-axis
+    (x_position - x).abs
+  end
+
+  def y_distance(y) #calculates how many tiles piece has moved on y-axis
+    (y_position - y).abs
+  end
+
+  def moved?(x,y) #validates whether piece is at its origin position
+    x != x_position || y != y_position
+  end
+
+  def horizontal_or_vertical_move?(x,y) #validates a move any amount of spaces along one axis
+    (x_distance(x) == 0 || y_distance(y) == 0)
+  end
 
-# Sample piece creation - Piece.create(type:"Queen", color:"white") type must be capitalized.
-                      # - Rook.create(color:"white")
\ No newline at end of file
+  def diagonal_move?(x,y) #validates a move any amount of spaces along a diagonal
+    x_distance(x) == y_distance(y)
+  end
+
+  def radial_move?(x,y) #validates a move 1 space in any direction
+    x_distance(x) < 2 && y_distance(y) < 2
+  end
+
+  def forward_move?(x,y) #validates a move 1 space forward
```

Comment:
So, I struggled with this as far as code organization.  The way I looked at it, I had some methods that were shared between the pieces related to their moves, and some that weren't.  But instead of mix and matching, I put ALL the move logic in Piece and then built the final validation function in the subclasses.  I was aware, and annoyed, the downside of this is that certain methods you expected to be in, for example, Pawn.rb, were now in Piece.rb, but I felt like I was choosing between two bad choices because you'd still have moments where you are not sure whether logic is in the subclass or parent class.

So given the ups and downs, I'm assuming the standard is to do what you suggested and only put stuff in Piece.rb that is shared by at least two subclasses, and deal with the tradeoffs?